### PR TITLE
Fix and improve shopping cart

### PR DESCRIPTION
**Pull request by ghost**
- Fix login page redirect when putting product in cart
- Products ordered by item name (after weightage)
- Fix shopping cart crash when removing the last product from cart
- Don't display tax in cart if amount is zero
- Translations

shopping-cart

### CODE REVIEW

**Comment by rmehta** [comment]
@britlog thanks for the PR. Can you add an animated GIF to show that this works. Since the users are concerned about stability, we want to ensure that user testing is done at your end. You can use LiceCAP to take an animated GIF.

**Comment by britlog** [comment]
@rmehta Done, but I can't reopen it because it was not closed by myself, could you please do it ? Thanks

**Comment by rmehta** [comment]
Thanks! The strawberries look delicious 🍓

File: erpnext/public/js/shopping_cart.js [hunk 1]
```diff
@@ -36,7 +36,6 @@ $.extend(shopping_cart, {
 	},
 
 	update_cart: function(opts) {
-		var full_name = frappe.session.user_fullname;
```
Comment (britlog) [inline erpnext/public/js/shopping_cart.js]: I 've noticed frappe.session.user_fullname was empty, so i just revert to the last version which worked but indeed use global. Thanks for review.

File: erpnext/public/js/shopping_cart.js [hunk 2]
```diff
@@ -35,10 +35,8 @@ $.extend(shopping_cart, {
 		});
 	},
 
-	update_cart: function(opts) {
-		var full_name = frappe.session && frappe.session.user_fullname;
-		
-		if(!full_name || full_name==="Guest") {
+	update_cart: function(opts) {		
+		if(fraappe.session.user==="Guest") {
```
Comment (britlog) [inline erpnext/public/js/shopping_cart.js]: There is a typing error on frappe word  ` if(fraappe.session.user==="Guest")`

**Comment by rmehta** [comment]
Thanks fixed.

**Review by sophatvathana — COMMENTED** [review]
Ok

**Pull request merged**